### DISCUSSION TITLE
How can I only float small windows and dialog boxes?

### DISCUSSION THREAD

**Discussion by BeansIsFat** [asmvik/yabai#1300] [Q&A]
Is there a way to always float windows and dialog boxes that are under a certain size? I would expect this to be done with a rule don't see anything in the rules documentation that mentions window size. And I haven't found anything in the wiki or discussions that mention this.

**Comment by marcusmyers** [comment]
I set the manage to off on my child windows of my most popular apps. For example:

`yabai -m rule --add app="^Twitter$" title="Media" manage=off`

I've never looked at the possibility of a certain window size, but maybe the above can help you?

**Reply by BeansIsFat** [reply]
This is what I'm doing now, but as a new user of yabai it feels like a game of whack-a-mole. For example, here's my entry for Finder:

`yabai -m rule --add app="^Finder$" title="Info$|^Connect to Server$|^Connecting to|^Preparing to Copy$|^Copying" manage=off`

**Comment by MuhammedZakir** [comment]
Haven't tried/tested this, but should be doable by executing a shell script on `window_created` event:

`yabairc`:
```sh
yabai -m signal --add event=window_created \
    action=/path/to/below_shell_script.sh
```

shell script:
```sh
#  Contains dimensions of the window in `frame`.
local window=$(yabai -m query --windows --window)

# -e flag sets exit status to 0 if last output value is neither `false` nor `null`.
if echo "$window" | jq -e '.frame | (.w <= 500) and (.h <= 500)'
then
  yabai -m window --toggle float
fi
```

**Reply by BeansIsFat** [reply]
Thanks for putting in some thought to this. I gave it a try and I'm getting unexpected results. The new window gets resized and positioned as if it were going in the tree, but it ends up floating.

Curiously, this seems to happen to all windows, not just those under 500 x 500. I tried changing the `.x` and `.y` to `.w` and `.h` but that didn't seem to make a difference.

**Reply by MuhammedZakir** [reply]
That was a mistake. It is actually `w` and `h`. Corrected it in the script.

Have you tried changing `500` to some other value? I just used a random size.

>  The new window gets resized and positioned as if it were going in the tree, but it ends up floating.

Thinking about it again, maybe you will have to go the opposite way. That is, float new windows by default, and then using `window_created` event, tile windows with size greater `H x W`.

Can't give more help without testing myself, which I can't do for now.

**Reply by BeansIsFat** [reply]
I've tried different sizes but the result was the same. I also tried what you suggested by doing the following:

> float new windows by default

I used `yabai -m rule --add app='.+' manage=off` to do this and eliminated any other rules to avoid confusion while troubleshooting. Please let me know if there's a better way to keep bsp enabled but float new windows by default.

>  tile windows with size greater H x W

I changed the script to the following:

```
#!/bin/bash
#  Contains dimensions of the window in `frame`.
local window=$(yabai -m query --windows --window)

# -e flag sets exit status to 0 if last output value is neither `false` nor `null`.
if echo "$window" | jq -e '.frame | (.w > 400) and (.h > 400)'
then
  yabai -m window --toggle float
fi
```

In theory, this should create all windows as floating. And if the window is less that 400 x 400 it should toggle to tiled.

In practice, it starts out creating floating windows and sometimes creates tiled windows. I haven't been able to figure out why the behavior changes yet.
